external_executors/local_cromwell/local_cromwell.py

```diff
@@ -0,0 +1,255 @@
+    else:
+        localized_inputs_dict[entry] = value
+json.dump(localized_inputs_dict, open("localized_inputs.json", "w"))
+END
+
+# Record an inputs localized event
+record_event "InputsLocalized"
+
+# Run Cromwell
+record_event "CromwellStarting"
+set +e
+java8 -jar cromwell-30.1.jar run workflow.wdl --inputs localized_inputs.json 2>&1
+cromwell_rc=$?
+set -e
+record_event "CromwellFinished" "$cromwell_rc"
+
+# Record a log event
+python3 << END
+import boto3, datetime, glob, json, uuid
+event_id = str(uuid.uuid4())
+ddb = boto3.resource("dynamodb", region_name="{region}")
+table = ddb.Table("{db_table}")
+try:
+    log_path = glob.glob("cromwell-workflow-logs/*log")[0]
+    log = json.dumps(open(log_path).read())
+except IndexError:
+    log = "No Cromwell logs found."
+    
+table.put_item(
+    Item={{
+        "EventId": event_id,
+        "JobId": "{job_id}",
+        "EventType": "JobLog",
+        "Message": log,
+        "Time": "{{:%Y-%m-%dT%H:%M:%SZ}}".format(datetime.datetime.utcnow())
+    }}
+)
+END
+
+if [ $cromwell_rc == 0 ]; then
+    record_event "JobSucceeded"
+else
+    record_event "JobFailed"
+fi
+
+trap - EXIT
+shutdown -H 5
+"""
+
+# Define boto3 clients
+EC2 = boto3.client('ec2', region_name=REGION)
+
+def workflows_post(event, context):
+    """Handle a POST to /workflows by creating a job.
+
+    The event should contain the following inputs, based off of GA4GH's WES:
+        - workflow_descriptor
+            WDL string that describes the workflow that will be run.
+        - workflow_params
+            JSON string that has the inputs for WDL
+        - workflow_type
+            Ignored, we're only going to try to run WDLs
+        - workflow_type_version
+            Ignored
+        - workflow_dependencies
+            (Optional) base64 encoded ZIP file with WDL dependencies
+    """
+
+    print(event)
+    print(context)
+
+    ec2_profile = os.environ["IAM_PROFILE"]
+    db_table = os.environ["DB_TABLE"]
+
+    # Create a unique id for this job that. We'll pass it to the EC2 instance
+    # that runs the job itself so it can use it to write status updates to the
+    # DB.
+    job_id = str(uuid.uuid4())
+
+    try:
+        workflow_dependencies = event["workflow_dependencies"]
+    except KeyError:
+        workflow_dependencies = ""
+
+    run_script = RUN_SCRIPT.format(
+        inputs_json=shlex.quote(event["workflow_params"]),
+        wdl=shlex.quote(event["workflow_descriptor"]),
+        dependencies=workflow_dependencies,
+        job_id=job_id,
+        db_table=db_table,
+        region=REGION)
+
+    print(run_script)
+    instance = EC2.run_instances(
+        ImageId=AMI,
+        InstanceType=INSTANCE_TYPE,
+        MinCount=1,
+        MaxCount=1,
+        InstanceInitiatedShutdownBehavior='terminate',
+        UserData=run_script,
+        IamInstanceProfile={
+            "Arn": ec2_profile
+        }
+    )
+
+    #instance_id = instance['Instances'][0]['InstanceId']
```

Comment:
Is this residual or meant to inform how to get the instance ID? If the latter, maybe be more verbose in your comment.

Reply:
This is an old debugging line. I'll remove it.